### DISCUSSION TITLE
How can I know orchard is about to start a new contentItem creation?

### DISCUSSION THREAD

**Discussion by AmadeuAntunes** [OrchardCMS/OrchardCore#8653] [Q&A] [answered]
I'm tryng to use the ContentHandlerBase and trying to get I'm about to start field a new content.
I don't want event save but when open the  new page to start field the fields.

I'm trying to use **ActivatingAsync** but seems the context don't brings me the content item empty or field with open current content. 
```
public override Task ActivatingAsync(ActivatingContentContext context)
{
 **Would like to get the contentItem  depending of contentITem type set a value for a field.**

  return base.ActivatingAsync(context);
}
```

The main idea is to set default values on initializing the form.

**Comment by Skrypt** [comment]
Need to think about it first but it seems evident that we are missing the current persisted contentItem here before submitting any changes on it. We have it in a different event, need to see why we don't have it here in this one.

**Reply by AmadeuAntunes** [reply]
Is it normal for ActivatingAsync execute more then one time when loading a page of a content item ?

**Reply by Skrypt** [reply]
It should do a ActivatingAsync only once per content item normally. So here, when you display a page it should execute only once unless it's initated a second time from your controller.

**Reply by Skrypt** [reply]
Best way to compare is to use a clean source code solution and add your Handler in there.

**Reply by AmadeuAntunes** [reply]
when I try load https://localhost:44351/Contents/ContentTypes/Email/Create it execute more then 2 times

**Reply by Skrypt** [reply]
Do you have nested content types? ListParts, BagParts etc ...

**Reply by AmadeuAntunes** [reply]
I have ContentParts added there

**Reply by AmadeuAntunes** [reply]
![image](https://user-images.githubusercontent.com/47931101/108748118-225aec80-7536-11eb-8837-d60d3e7cf9ee.png)

![image](https://user-images.githubusercontent.com/47931101/108748202-3b639d80-7536-11eb-94d6-bf6db86f3700.png)

**Reply by Skrypt** [reply]
That might be the reason because we have a ContentPartHandler that inherits from ContentHandlerBase.

**Reply by AmadeuAntunes** [reply]
but the ClonedAsync is working ok, brings me the CloneContentItem and ContentItem

**Reply by AmadeuAntunes** [reply]
![image](https://user-images.githubusercontent.com/47931101/108753853-2a6a5a80-753d-11eb-923e-aed3f8899cf0.png)
![image](https://user-images.githubusercontent.com/47931101/108753963-466dfc00-753d-11eb-823e-e0e08e45a629.png)
![image](https://user-images.githubusercontent.com/47931101/108754017-54238180-753d-11eb-9da9-b90c87a597d2.png)
![image](https://user-images.githubusercontent.com/47931101/108754086-6a314200-753d-11eb-9ab1-7421a2dd3962.png)

**Reply by Skrypt** [reply]
ClonedAsync makes sense but try InitializingAsync, LoadingAsync or UpdatingAsync / UpdatedAsync to compare them.

**Reply by AmadeuAntunes** [reply]
this test was done in a cleaned solutions

**Reply by AmadeuAntunes** [reply]
InitializingAsync when open the contentItem Test page on edit
![image](https://user-images.githubusercontent.com/47931101/108756478-75d23800-7540-11eb-980c-1b7fb18fb5f5.png)

InitializingAsync in customized solutions seems bring some info from ContentParts but not the current item when load editing page

LoadingAsync doesn't get executed when tried to create new item

In customized solutions the LoadingAsync  get executed and brings me some ContentParts info

**Comment by deanmarcussen** [comment, marked as answer]
`InitializingAsync` has worked for me previously to pre populate some content that is being created.

I think that was what you are trying to achieve?

Activiating isn't quite the right one for that

**Reply by AmadeuAntunes** [reply]
I've tried the InitializingAsync but seems it get call more then one time when the create page is loaded. should it get called only one time isn't?

**Reply by AmadeuAntunes** [reply]
![image](https://user-images.githubusercontent.com/47931101/109017758-61f91400-76af-11eb-84b8-ff39fdc8e390.png)

I tried this way to initialize values but isn't working the values don't goes into forms.

**Reply by AmadeuAntunes** [reply]
![image](https://user-images.githubusercontent.com/47931101/109021916-3c6e0980-76b3-11eb-91c8-99b0431a38d0.png)

I used this method now it worked but only work with forms already created and with it I was be able to replace the database data with this code.

**Reply by deanmarcussen** [reply]
You need to use the Content Item Extensions to modify data like this.

see `.Apply` and `.Alter`

There are many discussions here were we have answered how to use them
